src/SpecialCase/ComposerJsonZendFrameworkPackageSpecialCase.php

```diff
@@ -0,0 +1,699 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-migration for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-migration/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-migration/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\Migration\SpecialCase;
+
+use Composer\Semver\Semver;
+use RuntimeException;
+
+class ComposerJsonZendFrameworkPackageSpecialCase implements SpecialCaseInterface
+{
+    public function matches($filename, $content)
+    {
+        return (bool) preg_match('#\bcomposer\.json$#', $filename)
+            && null !== strpos($content, '"zendframework/zendframework": "');
```

Comment:
I would prefer here check if we have it set in require section, the above probably works for 99% but there might be very edge case where it will be wrong.